lib/dor/text_extraction/speech_to_text.rb

```diff
@@ -32,6 +32,21 @@ def required?
         workflow_context['runSpeechToText'] || false
       end
 
+      # remove any files in S3 workspace that are no longer needed
+      def cleanup
+        cleanup_s3_folder
+        true
+      end
+
+      # remove all files in the s3 input folder, the output folder is a subfolder, so gets cleaned up too
+      def cleanup_s3_folder
+        # Iterate over the list of filenames to be deleted (which is based on the job_id prefix for this druid)
+        s3_objects = aws_provider.client.list_objects(bucket: aws_provider.bucket_name, prefix: job_id)
```

Comment:
to deal with the corner case mentioned in the ticket:
> Note: The clean up should be careful to to list objects with the prefix ending in / because listing keys for /media/bb222cc3333-v2 would include /media/bb222cc3333-v20 ; which is kind of counter intuitive if you start thinking of s3 like a filesystem

should prefix actually be `#{job_id}/`?

Reply:
but also, will multiple versions ever be in flight at the same time?  the version is just getting passed around because it makes the workflow service calls easier, right?

Reply:
yes, thanks -- good catch

i fixed in this PR by adding the trailing slash, verified with specs and running on the console to be sure it still worked as expected